solr/solrj/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -540,7 +627,7 @@ private void constructState(Set<String> changedCollections) {
       log.debug(
           "clusterStateSet: interesting [{}] watched [{}] lazy [{}] total [{}]",
           collectionWatches.keySet().size(),
-          watchedCollectionStates.keySet().size(),
+          collectionWatches.activeCollections().size(),
```

Comment:
I would create a specialized method for this that doesn't involve collecting into an intermediate set.

Reply:
Thank you for the suggestion @madrob ! Yes, I struggled a little bit on this one as I first created a separate set to keep track of active collections, but once there are multiple collections, there are sync issues. As for a specialized method, can you give a bit more details on what it will look like?

Reply:
```
private long activeSize() {
       return this.entrySet().stream()
           .filter(entry -> entry.getValue().currentState != null)
           .count();
}
```

Reply:
Thanks @madrob ! There's a if `(log.isTraceEnabled()) {` later on that prints out all the names. So are you proposing to have 2 methods, one for count and one for the active set of names?

Reply:
You already have a method that gets all the names (activeCollections). Having two methods for different things is fine, even if one can be implemented in terms of the other because we are able to make it so much cleaner for the more common use case.